src/components/ProductFilters.tsx

```diff
@@ -1,103 +1,117 @@
 import { useAppDispatch, useTypedSelector } from "../store";
 import {
-  filterProductsByPrice,
-  filterProductsBySize,
-  sortProductsByType,
-  updateChosenSizes,
-} from "../store/slices/productList.slice";
+  setSizeFilter,
+  setPriceFilter,
+  setSortType,
+  sortByTypeAction,
+  filterByPriceAction,
+  filterBySizeAction,
+} from "../store/slices/productFilters";
 import { setGloablLoading } from "../store/slices/global.slice";
 import Form from "react-bootstrap/Form";
-import { Col, Container } from "react-bootstrap";
+import { Button, Col, Container, Row } from "react-bootstrap";
+import { singleProduct } from "../mock-data";
+import { productSortTypes } from "../store/slices/productFilters/model";
 
-const selectSortOptions = [
-  { type: "low_First", name: "low price first" },
-  { type: "high_First", name: "high price first" },
-];
 export type ProductFiltersProps = {};
 
 const ProductFilters: FC<ProductFiltersProps> = () => {
-  const filter = useTypedSelector((state) => state.productList.productFilters);
-  const [maxPrice, setMaxPrice] = useState<number>(filter.price.max);
   const dispatch = useAppDispatch();
+  const {filterOptions} = useTypedSelector((state) => state.productFilters);
+  const [activeButton, setActiveButton] = useState<boolean>(false);
+  const [maxPrice, setMaxPrice] = useState<number>(100);
+  const [minPrice, setMinPrice] = useState<number>(1);
 
-  const handleChangeSort = (e: ChangeEvent<HTMLSelectElement>) => {
-    dispatch(sortProductsByType(e.target.value));
-  };
-  const handleChandeMaxPrice = (e: ChangeEvent<HTMLInputElement>) => {
-    const currMaxPriceValue = Number(e.target.value);
-    if (currMaxPriceValue < 1) return;
-    setMaxPrice(currMaxPriceValue);
+  const handlePriceUpdate = (e: ChangeEvent<HTMLInputElement>) => {
+    const valueMax = Number(e.target.value);
+    setMaxPrice(valueMax);
+    // setMinPrice(value)
+    setActiveButton(true);
   };
-  const handleMouseUp = () => {
-    dispatch(setGloablLoading(true));
-
-    setTimeout(() => {
-      dispatch(filterProductsByPrice(maxPrice));
-      dispatch(setGloablLoading(false));
-    }, 500);
+  const handleSliderMouseUp = () => {
+    dispatch(setPriceFilter({ max: maxPrice, min: minPrice }));
   };
   const handleUpdateCheckBox = (e: ChangeEvent<HTMLInputElement>) => {
-    dispatch(updateChosenSizes(e.target.value));
+    dispatch(setSizeFilter(e.target.value));
+    setActiveButton(true);
+  };
+  const handleChangeSort = (e: ChangeEvent<HTMLSelectElement>) => {
+    dispatch(setSortType(e.target.value as productSortTypes));
+    setActiveButton(true);
   };
 
-  useEffect(() => {
-    // console.log(filter.sizes.chosen);
-    // console.log(filter.sizes.available);
-    dispatch(filterProductsBySize());
-  }, [filter.sizes.chosen]);
+  const aplyFilters = () => {
+    console.log(filterOptions);
+    setActiveButton(false);
+    dispatch(setGloablLoading(true));
+    setTimeout(() => dispatch(setGloablLoading(false)), 300);
```

Comment:
Incorrect works: loader displaying but list already updated. User shouldn't see loader and new updated list at the same time.
For ex: update setTimeout value to 2000 and try it

Reply:
+